src/utils/index.ts

```diff
@@ -1,196 +1,64 @@
-    .reduce(
-      ([_ = [], ...rest], username) =>
-        _.length < 100 ? [[..._, username], ...rest] : [[username], _, ...rest],
-      []
-    )
-    .reverse();
-
-  const userIdQueryStrings = arraysOfUsernames.map(
-    usernames => `login=${usernames.join("&login=")}`
-  );
-
-  for (const query of userIdQueryStrings) {
-    try {
-      const { data: profiles = [] } = await callTwitchApi(
-        `https://api.twitch.tv/helix/users?${query}`,
-        {}
-      );
-
-      profiles.forEach((profile: TwitchProfile) => {
-        twitchProfiles.push(addProfileToCache(profile.login, profile));
-      });
-    } catch (e) {
-      BeastieLogger.warn(`Failed to fetch twitch user ${query}`);
-    }
-  }
-  return twitchProfiles;
+  return await response.json();
 }
 
-const getChatroomViewers = async (): Promise<TwitchProfile[]> => {
-  let chatters: ChattersData;
-  try {
-    chatters = (
-      await callTwitchApi(
-        `https://tmi.twitch.tv/group/user/${config.BROADCASTER_USERNAME}/chatters`,
-        {}
-      )
-    ).chatters;
-  } catch (e) {
-    BeastieLogger.warn(`Failed to get chatroom viewers: ${e}`);
-    return [];
-  }
-
-  return getTwitchProfiles(Object.values(chatters).flat());
-};
-
-export const updateTeammateAwesomeness = async (user, amount) => {
-  const twitchUser = await getTwitchProfile(user);
+export const isGuildMaster = name =>
+  name === config.DISCORD_GUILD_MASTER_USERNAME;
+
+export const updateTeammateAwesomeness = async (
+  user,
+  amount
+): Promise<string> => {
+  let profile: TwitchProfile = await tryCatchLog<TwitchProfile>(
+    async () => getTwitchProfile(user),
+    BeastieLogger.warn,
+    `Failed to get twitch profile for ${user}`
+  );
 
-  if (!twitchUser) {
+  if (!profile) {
     return `I cannot find that username...`;
   }
 
-  if (!(await updateAwesomeness(twitchUser.id, twitchUser.login, amount))) {
+  if (!(await updateAwesomeness(profile.id, profile.login, amount))) {
     return `I failed to update the awesomeness :(`;
   }
 
   BeastieLogger.info(
-    `AWESOMENESS: ${twitchUser.display_name} received ${amount} awesomeness`
+    `AWESOMENESS: ${profile.display_name} received ${amount} awesomeness`
   );
-  return `Awarded ${twitchUser.display_name} ${amount} Awesomeness!`;
+  return `Awarded ${profile.display_name} ${amount} Awesomeness!`;
 };
 
-export const updateChattersAwesomeness = async amount => {
-  let viewers: TwitchProfile[];
-  try {
-    viewers = await getChatroomViewers();
-  } catch (e) {
-    BeastieLogger.warn(`Failed to get chatroom viewers!`);
-    return;
+export const updateChattersAwesomeness = async (
+  amount: number
+): Promise<string> => {
+  if (isNaN(amount)) {
+    return `I don't think that is a number o.o`;
   }
+
+  let viewers: TwitchProfile[] = await tryCatchLog<TwitchProfile[]>(
+    async () => getTwitchChatters(config.BROADCASTER_USERNAME),
+    BeastieLogger.warn,
+    "Failed to get chatroom viewers"
+  );
```

Comment:
```suggestion
  let viewers: TwitchProfile[] = await getTwitchChatters(config.BROADCASTER_USERNAME)
    .catch(swallowRejection("Failed to get chatroom viewers", BeastieLogger.warn));
```
See ➡️ https://github.com/teamTALIMA/BeastieBot-V3/pull/13/files#r429028595